opencga-catalog/src/main/java/org/opencb/opencga/catalog/managers/ClinicalAnalysisManager.java

```diff
@@ -1928,6 +1936,71 @@ public OpenCGAResult<?> updateCvdbIndex(String studyFqn, ClinicalAnalysis clinic
+            case CvdbIndexStatus.NONE:
+                if (!Arrays.asList(CvdbIndexStatus.PENDING_INDEX, CvdbIndexStatus.READY, CvdbIndexStatus.ERROR).contains(newStatus)) {
+                    throw new CatalogException("CVDB Index status cannot be changed from NONE to " + newStatus);
+                }
+                break;
+            case CvdbIndexStatus.PENDING_INDEX:
+                if (!Arrays.asList(CvdbIndexStatus.QUEUED_INDEX, CvdbIndexStatus.ERROR, CvdbIndexStatus.READY, CvdbIndexStatus.NONE,
+                        CvdbIndexStatus.PENDING_REMOVE).contains(newStatus)) {
+                    throw new CatalogException("CVDB Index status cannot be changed from PENDING_INDEX to " + newStatus);
+                }
+                break;
+            case CvdbIndexStatus.PENDING_OVERWRITE:
+                if (!Arrays.asList(CvdbIndexStatus.QUEUED_OVERWRITE, CvdbIndexStatus.ERROR, CvdbIndexStatus.READY,
+                        CvdbIndexStatus.PENDING_REMOVE).contains(newStatus)) {
+                    throw new CatalogException("CVDB Index status cannot be changed from PENDING_OVERWRITE to " + newStatus);
+                }
+                break;
+            case CvdbIndexStatus.PENDING_REMOVE:
+                if (!Arrays.asList(CvdbIndexStatus.QUEUED_REMOVE, CvdbIndexStatus.ERROR, CvdbIndexStatus.READY, CvdbIndexStatus.NONE,
+                        CvdbIndexStatus.PENDING_INDEX).contains(newStatus)) {
+                    throw new CatalogException("CVDB Index status cannot be changed from PENDING_REMOVE to " + newStatus);
+                }
+                break;
+            case CvdbIndexStatus.QUEUED_INDEX:
+                if (!Arrays.asList(CvdbIndexStatus.ERROR, CvdbIndexStatus.READY, CvdbIndexStatus.NONE, CvdbIndexStatus.PENDING_REMOVE)
+                        .contains(newStatus)) {
+                    throw new CatalogException("CVDB Index status cannot be changed from QUEUED_INDEX to " + newStatus);
+                }
+                break;
+            case CvdbIndexStatus.QUEUED_OVERWRITE:
+                if (!Arrays.asList(CvdbIndexStatus.ERROR, CvdbIndexStatus.READY, CvdbIndexStatus.PENDING_REMOVE).contains(newStatus)) {
+                    throw new CatalogException("CVDB Index status cannot be changed from QUEUED_OVERWRITE to " + newStatus);
+                }
+                break;
+            case CvdbIndexStatus.QUEUED_REMOVE:
+                if (!Arrays.asList(CvdbIndexStatus.NONE, CvdbIndexStatus.ERROR, CvdbIndexStatus.READY, CvdbIndexStatus.PENDING_INDEX)
+                        .contains(newStatus)) {
+                    throw new CatalogException("CVDB Index status cannot be changed from QUEUED_REMOVE to " + newStatus);
+                }
+                break;
+            case CvdbIndexStatus.READY:
+                if (!Arrays.asList(CvdbIndexStatus.ERROR, CvdbIndexStatus.PENDING_REMOVE, CvdbIndexStatus.PENDING_OVERWRITE)
```

Comment:
Why can't change to pending_remove or pending_overwrite, but it's ok to change to pending_index? What's the logic of this?

Reply:
The state diagram of CvdbIndexStatus is somehow convoluted..